src/main/java/org/folio/dew/repository/BaseFilesStorage.java

```diff
@@ -0,0 +1,455 @@
+              .destinationKey(path)
+              .partNumber(1).build();
+
+            var uploadPartRequest2 = UploadPartRequest.builder()
+              .bucket(bucket)
+              .key(path)
+              .uploadId(uploadId)
+              .partNumber(2).build();
+
+            var originalEtag  = s3Client.uploadPartCopy(uploadPartRequest1).copyPartResult().eTag();
+            var appendedEtag = s3Client.uploadPart(uploadPartRequest2, RequestBody.fromBytes(bytes)).eTag();
+
+            var original = CompletedPart.builder()
+              .partNumber(1)
+              .eTag(originalEtag).build();
+            var appended = CompletedPart.builder()
+              .partNumber(2)
+              .eTag(appendedEtag).build();
+
+            var completedMultipartUpload = CompletedMultipartUpload.builder()
+              .parts(original, appended)
+              .build();
+
+            var completeMultipartUploadRequest =
+              CompleteMultipartUploadRequest.builder()
+                .bucket(bucket)
+                .key(path)
+                .uploadId(uploadId)
+                .multipartUpload(completedMultipartUpload)
+                .build();
+
+            s3Client.completeMultipartUpload(completeMultipartUploadRequest);
+
+          } else {
+
+            var temporaryFileName = path + "_temp";
+            write(temporaryFileName, bytes);
+
+            client.composeObject(ComposeObjectArgs.builder()
+              .bucket(bucket)
+              .region(region)
+              .object(path)
+              .sources(List.of(ComposeSource.builder()
+                  .bucket(bucket)
+                  .region(region)
+                  .object(path)
+                  .build(),
+                ComposeSource.builder()
+                  .bucket(bucket)
+                  .region(region)
+                  .object(temporaryFileName)
+                  .build()))
+              .build());
+
+            delete(temporaryFileName);
+
+          }
+
+        } else {
+          var original = readAllBytes(path);
+          byte[] composed = new byte[original.length + bytes.length];
+          System.arraycopy(original, 0, composed, 0, original.length);
+          System.arraycopy(bytes, 0, composed, original.length, bytes.length);
+          write(path, composed);
+        }
+      }
+    } catch (Exception e) {
+      throw new IOException("Cannot append data for path: " + path, e);
+    }
+  }
+
+  /**
+   * Deletes a file
+   *
+   * @param path - the path to the file to delete
+   * @throws FileOperationException if an I/O error occurs
+   */
+  public void delete(String path) {
+    try {
+      var paths = walk(path).collect(Collectors.toList());
+
+      var sb = new StringBuilder();
+
+      paths.forEach(p -> {
+        try {
+          client.removeObject(RemoveObjectArgs.builder()
+            .bucket(bucket)
+            .region(region)
+            .object(p)
+            .build());
+        } catch (Exception e) {
```

Comment:
I think better to add error message in log to understand why we could not delete file